### DISCUSSION TITLE
Terminal cursor changed after renaming a file.

### DISCUSSION THREAD

**Discussion by Ionizing** [sxyazi/yazi#1126] [1. Q&A] [answered]
`yazi` is a really fancy tool and help a lot on my work. Thanks for the outstanding job.

I usually use this tool on a ssh session combined with Tmux terminal multiplexer. I came with a situation that after renaming a file by pressing `r`, the terminal cursor of current terminal tab (not just current tmux window), changed to blinking line with the same style inside the yazi's renaming dialogue. I wonder if this is an exptected feature?

![image](https://github.com/sxyazi/yazi/assets/17506747/3a7571ce-d70f-4bd8-ac77-91c3be5c83d6)
![image](https://github.com/sxyazi/yazi/assets/17506747/7e54969a-707d-4dc6-b1f6-f678acaebf80)

(Just ignore the `>` and `$`, since the prompt has nothing to do with the cursor)

My environment:
```
macOS 13 13.5.1 (ssh client)
RockyLinux 8.7 (ssh server with Tmux)
yazi 0.2.5 (VERGEN_IDEMPOTENT_OUTPUT 2024-04-23) (musl)
WezTerm 20240212-074107-22f9f8d2
```

**Comment by sxyazi** [comment]
Yeah this is expected, when the input is in insert mode, Yazi will use a "bar" shape instead of a "block" to indicate the current mode. 

As for the cursor blinking, there is a related issue here https://github.com/sxyazi/yazi/issues/1101, I'll find time to investigate it

**Reply by Ionizing** [reply]
Thanks for the response. The issue you posted above is somewhat not consistent with my situation. I mean, after renaming the file, and quitting `yazi`, the cursor for current WezTerm tab is changed to the blinking line, until I detach the tmux pane and close the ssh session. I understand the "bar" shape indicate the insert mode, but I'm confused that if it is the desired behavior that `yazi` changed the cursor outside `yazi`.

**Reply by sxyazi** [reply]
I see your point and tried my WezTerm with tmux setup but couldn't reproduce it. Is the problem still without tmux on your side?

Yazi will restore the cursor shape on exiting, the code here https://github.com/sxyazi/yazi/blob/5f8c20e2b517faea53680f720553d20ec363e9ab/yazi-shared/src/term/term.rs#L80

**Reply by Ionizing** [reply]
Thanks for the reply. I'll prepare an MWE later.

**Comment by Ionizing** [comment]
I find this issue may be related to `tmux` since I can reproduce it on two different machines with the tmux configuration in the following:

```tmux
# remap prefix from 'C-b' to 'C-\'
unbind ^b
set-option -g prefix '^\'
bind-key '^\' send-prefix

set -g status-keys vi
setw -g mode-keys vi
bind h select-pane -L
bind j select-pane -D
bind k select-pane -U
bind l select-pane -R

bind | split-window -h -c "#{pane_current_path}"
bind - split-window -v -c "#{pane_current_path}"

set -g default-terminal "screen-256color"
set -ga terminal-overrides ",*256col*:Tc"
set -ga terminal-overrides '*:Ss=\E[%p1%d q:Se=\E[ q'
set-environment -g COLORTERM "truecolor"

set -g @logging-path "~/.tmux"
```

where the reproducible tmux versions are `next-3.3` and `3.4`.

**Reply by sxyazi** [reply, marked as answer]
remove `set -ga terminal-overrides '*:Ss=\E[%p1%d q:Se=\E[ q'`

**Reply by Ionizing** [reply]
I forget why I added this line to `.tmux.conf`, but commenting out it does solves the issue. Thanks for the patient reply. I'll reopen this discussion once I recall the purpose of this `terminal-overrides` ...
